### PR TITLE
Refresh connectivity proxy credentials

### PR DESCRIPTION
**Pull request by koala7659**
- Stop checking release before reconciliation and skipping if version is the same - this allows to refresh credential every time the reconciliation loop is executed. After latest chart fix - pods will not be restarted.  
- Refactoring of generation of Istio CA Secret, make separate http client only to request for CA
- Old code removal
- Unit test adjustments and many fixes
- Running goimports on the project

### CODE REVIEW

File: pkg/reconciler/instances/connectivityproxy/action.go [hunk 1]
```diff
@@ -44,35 +45,45 @@ func (a *CustomAction) Run(context *service.ActionContext) error {
...
-			return errors.Wrap(err, "Error during installation")
+		refresh := app != nil
+
+		if refresh {
+			context.Logger.Info("Reconciling component")
+		} else {
+			context.Logger.Info("Installing component")
+		}
+
+		if err := a.Commands.InstallOrRefresh(context, refresh); err != nil {
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/action.go]: Changed to Apply

File: pkg/reconciler/instances/connectivityproxy/command.go [hunk 2]
```diff
@@ -149,3 +141,38 @@ func (a *CommandActions) removeIstioSecrets(context *service.ActionContext) erro
...
+	}
+
+	repo := secrets.NewSecretRepo(namespace, targetClientSet)
+	return repo.SaveIstioCASecret(secretName, secretKey, ca)
+}
+
+func getIstioSecretCfg(config map[string]interface{}) (string, string, string, error) {
+	istioSecretName, ok := config["istio.secret.name"]
+
+	if !ok {
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/command.go:160]: This code was ported directly from former istioSecretCopy function. I just maintained original logic to not cause any regression. I can change it though, in a reasonable manner.

File: pkg/reconciler/instances/connectivityproxy/command.go [hunk 3]
```diff
@@ -149,3 +141,38 @@ func (a *CommandActions) removeIstioSecrets(context *service.ActionContext) erro
...
+		return "", "", "", errors.New("missing configuration value istio.secret.name")
+	}
+
+	istioNamespace, ok := config["istio.secret.namespace"]
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/command.go:164]: I agree, this is the same case as the one above, I will try to make it better anyway

File: pkg/reconciler/instances/connectivityproxy/connectivityclient/connectivityclient.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+type ConnectivityCAClient struct {
+	url    string
+	client *http.Client
+}
+
+func NewConnectivityCAClient(task *reconciler.Task) (*ConnectivityCAClient, error) {
+	configs := task.Configuration
+
+	url, ok := configs["global.binding.url"]
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/connectivityclient/connectivityclient.go:26]: Good catch, I will fix it.

File: pkg/reconciler/instances/connectivityproxy/connectivityclient/connectivityclient.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		return nil, errors.New("cannot create request")
+	}
+
+	resp, err := clientCA.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode == 404 {
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/connectivityclient/connectivityclient.go:60]: Let's agree on that!

File: pkg/reconciler/instances/connectivityproxy/connectivityclient/connectivityclient.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/connectivityclient/connectivityclient.go:26]: I will add a separate test for that function

File: pkg/reconciler/instances/connectivityproxy/action.go [hunk 6]
```diff
@@ -44,35 +45,45 @@ func (a *CustomAction) Run(context *service.ActionContext) error {
...
 
 		context.Logger.Debug("Service Binding Secret check")
-		if err != nil || bindingSecret == nil {
-			return errors.Wrap(err, "Error while retrieving secret")
+		if err != nil {
+			return errors.Wrap(err, "Error while retrieving service binding secret")
 		}
 
+		// TODO rethink binding secret retrieval
+		if bindingSecret == nil {
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/action.go:58]: The `FindSecret` function was not within scope of the refactoring task. I propose to change it during stage 2

File: pkg/reconciler/instances/connectivityproxy/command.go [hunk 7]
```diff
@@ -19,25 +22,19 @@ const (
...
 type CommandActions struct {
-	clientSetFactory       NewInClusterClientSet
-	targetClientSetFactory NewTargetClientSet
-	install                service.Operation
-	copyFactory            []CopyFactory
+	install service.Operation
 }
 
-func (a *CommandActions) InstallOrUpgrade(context *service.ActionContext, app *appsv1.StatefulSet) error {
+func (a *CommandActions) InstallOrRefresh(context *service.ActionContext, refresh bool) error {
```
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/command.go]: Changed to Apply

File: pkg/reconciler/instances/connectivityproxy/command.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (koala7659) [reply pkg/reconciler/instances/connectivityproxy/command.go:164]: Actually I fixed it according to your suggestion

**Review by akgalwas — APPROVED** [review]

**Pull request merged**